core/src/main/java/io/grpc/EquivalentAddressGroup.java

```diff
@@ -60,6 +61,33 @@ public EquivalentAddressGroup(SocketAddress addr) {
   }
 
   /**
+   * Converts list of ResolvedServerInfoGroup objects into one EquivalentAddressGroup object.
+   */
+  public static EquivalentAddressGroup fromResolvedServerInfoGroups(
```

Comment:
GrpclbLoadBalancer will soon be changed to not doing so. For the case of GRPCLB, an address group from NameResolver will carry a flag indicating whether it's load-balancer or not, and GrpclbLoadBalancer will be more sophisticated in handling these addresses.